File: nysol_python/mcmd/methods/convTex/db/smp/mnewnumber.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mnewnumber'

############################### IDATA
db['idatas']=[]

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
1から始まる間隔が1の連番をふり、 ``No.`` という項目名で新規に5行のデータを作成する。
'''
script['sh_code']='''
mnewnumber a=No. I=1 S=1 l=5 o=rsl1.csv
'''
script['py_code']='''
nm.mnewnumber(a="No.", I="1", S="1", l="5", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='開始番号と間隔の変更'
script['comment']='''
10から始まる間隔が5の連番をふり、 ``No.`` という項目名で新規に5行のデータを作成する。
'''
script['sh_code']='''
mnewnumber a=No. I=5 S=10 l=5 o=rsl2.csv
'''
script['py_code']='''
nm.mnewnumber(a="No.", I="5", S="10", l="5", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='アルファベット連番'
script['comment']='''
Aから始まる間隔が1のアルファベット連番をふり、 ``No.`` という項目名で新規に5行のデータを作成する。
'''
script['sh_code']='''
mnewnumber a=No. I=1 S=A l=5 o=rsl3.csv
'''
script['py_code']='''
nm.mnewnumber(a="No.", I="1", S="A", l="5", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='ヘッダ行なしで新規作成'
script['comment']='''
Bから始まる間隔が3のアルファベット連番をふり、ヘッダなしで新規に11行のデータを作成する。
'''
script['sh_code']='''
mnewnumber  -nfn  I=3 l=11 S=B o=rsl4.csv
'''
script['py_code']='''
nm.mnewnumber(nfn=True, I="3", l="11", S="B", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mnewnumber.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

############## Example 1
title="Basic Example"
comment=<<'EOF'
Generate a dataset with 5 sequential numbers starting from 1 incremented by 1. Name the sequence as \verb|No.|.
EOF
scp=<<'EOF'
mnewnumber a=No. I=1 S=1 l=5 o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Change the starting number and interval "
comment=<<'EOF'
Generate a dataset consisting of 5 sequential numbers starting from 10 with an incremental interval of 5. Name the sequence as \verb|No.|.
EOF
scp=<<'EOF'
mnewnumber a=No. I=5 S=10 l=5 o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Generate series of alphabet"
comment=<<'EOF'
Generate a dataset consisting of 5 alphabet sequence starting from A with 1 alphabet in between. Name the sequence as \verb|No.|.
EOF
scp=<<'EOF'
mnewnumber a=No. I=1 S=A l=5 o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## Example 4
title="Generate data without header"
comment=<<'EOF'
Generate a dataset consisting of 11 alphabet sequence starting from B with 3 alphabets in between. Exclude the header from the output.
EOF
scp=<<'EOF'
mnewnumber  -nfn  I=3 l=11 S=B o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
